plugin.json

```diff
@@ -21,75 +21,8 @@
-                "default": "words",
-                "options": [
-                    {
-                        "display_name": "Random English words in title case (e.g. PlayfulDragonsObserveCuriously)",
-                        "value": "words"
-                    },
-                    {
-                        "display_name": "UUID (universally unique identifier)",
-                        "value": "uuid"
-                    },
-                    {
-                        "display_name": "Mattermost context specific names. Combination of team name, channel name, and random text in Public and Private channels; personal meeting name in Direct and Group Message channels.",
-                        "value": "mattermost"
-                    },
-                    {
-                        "display_name": "Allow user to select meeting name",
-                        "value": "ask"
-                    }
-                ]
-            },
-            {
-                "key": "JitsiJWT",
-                "display_name": "Use JWT Authentication for Jitsi:",
-                "type": "bool",
-                "help_text": "(Optional) If your Jitsi server uses JSON Web Tokens (JWT) for authentication, set this value to true."
-            },
-            {
-                "key": "JitsiAppID",
-                "display_name": "App ID for JWT Authentication:",
-                "type": "text",
-                "help_text": "(Optional) The app ID used for authentication by the Jitsi server and JWT token generator."
-            },
-            {
-                "key": "JitsiAppSecret",
-                "display_name": "App Secret for JWT Authentication:",
-                "type": "text",
-                "help_text": "(Optional) The app secret used for authentication by the Jitsi server and JWT token generator."
-            },
-            {
-                "key": "JitsiLinkValidTime",
-                "display_name": "Meeting Link Expiry Time (minutes):",
-                "type": "number",
-                "help_text": "(Optional) The number of minutes from when the meeting link is created to when it becomes invalid. Minimum is 1 minute. Only applies if using JWT authentication for your Jitsi server.",
-                "default": 30
-            },
-            {
-                "key": "JitsiCompatibilityMode",
-                "display_name": "Enable Compatibility Mode:",
-                "type": "bool",
-                "help_text": "(Insecure) If your Jitsi server is not compatible with this plugin, include the JavaScript API hosted on your Jitsi server directly in Mattermost instead of the default API version provided by the plugin. **WARNING:** Enabling this setting can compromise the security of your Mattermost system, if your Jitsi server is not fully trusted and allows direct modification of program files. Use with caution.",
-                "default": false
+                "key": "JitsiSettings",
```

Comment:
@mickmister  No this is not backward compatible as we have changed the name of key of plugin settings. Here we have used this additional structure to use this with a custom component we have created and registered for getting the plugin config settings . That custom component renders different settings according to the selected option of either jitsi or jaas.It is registered in plugin.json file with key:- JitsiSettings and in webapp the same component can be found src/components/admin settings.
I dont know how we can achieve the same thing without using any addition struct in configurations.

Reply:
@Kshitij-Katiyar Please put backwards-compatibility at a top priority. We can just add the additional config settings as regular config settings instead of using custom settings. Only a subset of the settings will apply, depending on if JaaS is enabled or not.